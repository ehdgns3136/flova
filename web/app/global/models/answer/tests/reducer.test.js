
import { fromJS } from 'immutable';
import answerReducer from '../reducer';
import {
  openLikerModal,
  closeLikerModal,
  getAnswerLikersRequest,
  getAnswerLikersSuccess,
  getAnswerLikersFailure,
} from '../actions';
import { MOCK_USERS } from './mocks/index';


describe('answerReducer', () => {
  it('should return the initial state', () => {
    expect(answerReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle openLikerModal action correctly', () => {
    const state1 = fromJS({
      isLikerModalOpened: false,
    });

    const stateResult1 = fromJS({
      isLikerModalOpened: true,
    });

    expect(answerReducer(state1, openLikerModal())).toEqual(stateResult1);
  });

  it('should handle closeLikerModal action correctly', () => {
    const state1 = fromJS({
      isLikerModalOpened: true,
    });

    const stateResult1 = fromJS({
      isLikerModalOpened: false,
    });

    expect(answerReducer(state1, closeLikerModal())).toEqual(stateResult1);
  });

  it('should handle getAnswerLikersRequest action correctly', () => {
    const state1 = fromJS({
      likers: null,
      isLikersLoading: false,
    });

    const stateResult1 = fromJS({
      likers: null,
      isLikersLoading: true,
    });

    expect(answerReducer(state1, getAnswerLikersRequest())).toEqual(stateResult1);
  });

  it('should handle getAnswerLikersSuccess action correctly', () => {
    const state1 = fromJS({
      likers: null,
      isLikersLoading: true,
    });

    const stateResult1 = fromJS({
      likers: [
        1,
        2,
        3,
      ],
      isLikersLoading: false,
    });

    expect(answerReducer(state1, getAnswerLikersSuccess(MOCK_USERS))).toEqual(stateResult1);
  });

  it('should handle getAnswerLikersFailure action correctly', () => {
    const state1 = fromJS({
      likers: [
        1,
        2,
        3,
      ],
      isLikersLoading: true,
    });

    const stateResult1 = fromJS({
      likers: null,
      isLikersLoading: false,
    });

    expect(answerReducer(state1, getAnswerLikersFailure())).toEqual(stateResult1);
  });
});
