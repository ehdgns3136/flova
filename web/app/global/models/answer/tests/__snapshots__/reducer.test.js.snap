exports[`answerReducer should return the initial state 1`] = `
Object {
  "initialContent": null,
  "initialTitle": null,
  "isAnswerUploadError": false,
  "isAnswerUploading": false,
  "isLikerModalOpened": false,
  "isLikersLoading": false,
  "likers": null,
  "questionID": null,
}
`;
